### PR TITLE
avoid using shared strings table when creating ARGV/KEYS tables

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: patches/lua-v5.4.4.patch [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+--- a/lapi.c
++++ b/lapi.c
+@@ -534,6 +534,17 @@ LUA_API const char *lua_pushlstring (lua_State *L, const char *s, size_t len) {
+   return getstr(ts);
+ }
+ 
++LUA_API const char *lua_pushlongstring (lua_State *L, const char *s, size_t len) {
++  TString *ts;
++  lua_lock(L);
++  ts = luaS_createlngstrobj(L, len);
```
Comment (adiholden) [inline patches/lua-v5.4.4.patch:12]: This looks good, I would just test manually that the flow works ok with empty string

**Review by adiholden — DISMISSED** [review]

**Comment by romange** [comment]
Good catch

**Review by kostasrim — DISMISSED** [review]

File: patches/lua-v5.4.4.patch (hunk 1 above)
Comment (adiholden) [reply patches/lua-v5.4.4.patch:18]: Ah? how is this working? shouldnt you remove lines 18 and 19 with this change of if else?

Comment (romange) [reply patches/lua-v5.4.4.patch:18]: you are right, sorry. Changed

**Review by adiholden — APPROVED** [review]

**Pull request closed without merging**